src/wallet/wallet.cpp

```diff
@@ -3277,8 +3277,9 @@ bool CWallet::SelectCoinsGroupedByAddresses(std::vector<CompactTallyItem>& vecTa
 
     isminefilter filter = ISMINE_SPENDABLE;
 
-    // try to use cache for already confirmed anonymizable inputs, no cache should be used when the limit is specified
-    if(nMaxOupointsPerAddress != -1 && fAnonymizable && fSkipUnconfirmed) {
```

Comment:
I think I understand why this would have caused a bug but I don't understand what the specific bug was. Can you describe what was wrong and how you noticed this?